src/Abstractions/IMigration.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) InfinityFlow. All Rights Reserved.
+// Licensed under the Apache 2.0. See LICENSE file in the solution root for full license information.
+
+namespace InfinityFlow.Temporal.Migrator.Abstractions;
+
+/// <summary>
+/// Migration.
+/// </summary>
+public interface IMigration
+{
+    /// <summary>
+    /// Execute.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+    ValueTask ExecuteAsync(CancellationToken cancellationToken);
```

Comment:
I think it makes sense here to expose mechanism to register dependencies in a container. 

We should probably have a scoped container each migration.